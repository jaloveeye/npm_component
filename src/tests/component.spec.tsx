import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SampleComponent from '..';

test("component testing", () => {

    const component = renderer.create(<SampleComponent text="Herace" size={10} />);
    const testInstance = component.root;

    expect(testInstance.findByType(SampleComponent).props.text).toBe("Herace");

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("component testing2", () => {

    const component = renderer.create(<SampleComponent text="Herace" size={10} />);
    const testInstance = component.root;

    expect(testInstance.findByType(SampleComponent).props.size).toBe(10);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});