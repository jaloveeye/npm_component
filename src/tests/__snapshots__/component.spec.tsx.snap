// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component testing 1`] = `
Array [
  "Herace",
  <div>
     
    1
  </div>,
  <div>
     
    2
  </div>,
  <div>
     
    3
  </div>,
  <div>
     
    4
  </div>,
  <div>
     
    5
  </div>,
  <div>
     
    6
  </div>,
  <div>
     
    7
  </div>,
  <div>
     
    8
  </div>,
  <div>
     
    9
  </div>,
  <div>
     
    10
  </div>,
]
`;

exports[`component testing2 1`] = `
Array [
  "Herace",
  <div>
     
    1
  </div>,
  <div>
     
    2
  </div>,
  <div>
     
    3
  </div>,
  <div>
     
    4
  </div>,
  <div>
     
    5
  </div>,
  <div>
     
    6
  </div>,
  <div>
     
    7
  </div>,
  <div>
     
    8
  </div>,
  <div>
     
    9
  </div>,
  <div>
     
    10
  </div>,
]
`;
